### PR TITLE
[Intel GPU] Avoid target_link_libraries twice for torch_xpu_ops which will potentially cause multiple definition symbol linker error.

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #135320
* #135318
* #139026
* #136742
* #139025
* __->__ #139024

[Intel GPU] Avoid target_link_libraries twice for torch_xpu_ops which will potentially cause multiple definition symbol linker error.

### CODE REVIEW

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -1068,15 +1068,14 @@ if(USE_XPU)
...
-    target_link_libraries(torch_xpu PRIVATE torch_xpu_ops)
     if(MSVC)
       # Windows
       target_link_libraries(torch_xpu PRIVATE
       "-WHOLEARCHIVE:\"$<TARGET_FILE:torch_xpu_ops>\"")
     else()
       # Linux
       target_link_libraries(torch_xpu PRIVATE
-        "-Wl,--whole-archive,\"$<TARGET_FILE:torch_xpu_ops>\" -Wl,--no-whole-archive")
+        "-Wl,--whole-archive" torch_xpu_ops "-Wl,--no-whole-archive")
```
Comment (EikanWang) [inline caffe2/CMakeLists.txt:1078]: One more question, have you evaluated the impact?

**Review by EikanWang — APPROVED** [review]

File: caffe2/CMakeLists.txt (hunk 1 above)
Comment (etaf) [reply caffe2/CMakeLists.txt:1078]: Sorry, I don't get the `impact` means ? We can pass the CI.

Comment (EikanWang) [reply caffe2/CMakeLists.txt:1078]: `\"$<TARGET_FILE:torch_xpu_ops>\" ` should be the full path, while `torch_xpu_ops` is not. Any impact?

File: caffe2/CMakeLists.txt [hunk 2]
```diff
@@ -1068,15 +1068,14 @@ if(USE_XPU)
   if(NOT TARGET torch_xpu_ops)
     message(WARNING "Failed to include ATen XPU implementation target")
   else()
-    target_link_libraries(torch_xpu PRIVATE torch_xpu_ops)
     if(MSVC)
       # Windows
       target_link_libraries(torch_xpu PRIVATE
       "-WHOLEARCHIVE:\"$<TARGET_FILE:torch_xpu_ops>\"")
```
Comment (fengyuan14) [inline caffe2/CMakeLists.txt:1074]: Please use cmake target for windows as well to decide dependency between `torch_xpu` and `torch_xpu_ops`.

**Review by EikanWang — APPROVED** [review]
LGTM

**Review by fengyuan14 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Comment by EikanWang** [comment]
@xianxl , pls. ensure the ci signal is green

**Pull request closed without merging**